divintseg/_core.py

```diff
@@ -422,3 +422,151 @@ def similarity(
             community.
         """
         return 1.0 - self.dissimilarity(df_communities)
+
+
+def isolation(
+    df_communities: pd.DataFrame,
+    group_name: str,
+    by=None,
```

Comment:
I'm rethinking this a bit in light of the code below. Let's make `by` and `over` required arguments instead of optional. We can add optionality later if it becomes important, but for now let's make them required.